assets/js/googlesitekit/notifications/register-defaults.js

```diff
@@ -295,15 +324,19 @@ export function registerDefaults( notificationsAPI ) {
 
 			const getModuleState = async ( moduleSlug, datastoreSlug ) => {
 				// Check if the module connected and return early if not.
-				const isConnected =
-					select( CORE_MODULES ).isModuleConnected( moduleSlug );
+				const isConnected = await resolveSelect(
+					CORE_MODULES
+				).isModuleConnected( moduleSlug );
 				if ( ! isConnected ) {
 					return 'disconnected';
 				}
 
 				// If we are in the view only mode, we need to ensure the user can view the module
 				// and it is not in the recovering state. Return early if either of these is wrong.
 				if ( viewOnly ) {
+					// This ensures resolution for the canViewSharedModule() selectors.
+					await resolveSelect( CORE_MODULES ).getModules();
+
```

Comment:
Same here, we can move `await resolveSelect( CORE_MODULES ).getModules();` to the beginning of the function and call it unconditionally because it will be needed in `isModuleConnected` anyway. Plus, we need to combine it `isGatheringData` and `getRecoverableModules`.